redaxo/src/core/lib/login/user.php

```diff
@@ -50,7 +50,28 @@ public static function get(int $id): ?self
         return static::getInstance($id, static function (int $id) {
             $sql = rex_sql::factory()->setQuery('SELECT * FROM '.rex::getTable('user').' WHERE id = ?', [$id]);
 
-            return $sql->getRows() ? new static($sql) : null;
+            if ($sql->getRows()) {
+                $user = new static($sql);
+                static::addInstance('login_' . $user->getLogin(), $user);
```

Comment:
hmm ich glaub hier ist der `login_` prefix nicht korrekt

Reply:
oder willst du bewusst "über kreuz" die caches füllen?

Reply:
Im "ID cache" wird bereits beim getInstance() das Objekt gespeichert, hier wird das Objekt nochmal zusätzlich im "Login Cache" gespeichert.

Reply:
Ich bin mir nicht sicher, ob ich diesen Doppel-Instanz-Cache so machen würde.
Alternative wäre, in einer statischen Property separat das Login->ID-Mapping zu füllen, und die Instanzen nur per ID zu cachen.

Aber bin mir wie gesagt unsicher, was besser ist. Also falls ihr beide für das Doppel-Instanz-Cachen seid, passt das auch für mich.
Du müsstest aber `clearInstance` aus dem Trait hier überschreiben/erweitern, sodass `rex_user::clearInstance($id)` auch die Instanz aus dem Login-Cache wirft.

Reply:
Zu clearInstance überschreiben: Das machen wir zum Beispiel auch in rex_sql_table: 

https://github.com/redaxo/redaxo/blob/e38f1e9c6532f752824a854df0814a675fabc14b/redaxo/src/core/lib/sql/table.php#L12-L14
https://github.com/redaxo/redaxo/blob/e38f1e9c6532f752824a854df0814a675fabc14b/redaxo/src/core/lib/sql/table.php#L181-L189

Reply:
@gharlan Würdest du das Login->ID Mapping in der user Klasse hinzufügen oder den instance_pool_trait um eine Alias Funktion erweitern?

Reply:
Beides für mich ok.